src/system/process.py

```diff
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import subprocess
+import tempfile
+
+import psutil  # type: ignore
+
+
+class Process:
+    def __init__(
+        self
+    ):
+        self.process = None
+        self.stdout = None
+        self.stderr = None
+        self.has_started = False
```

Comment:
You should call `self.started` and in the method check `self.process` thus avoiding another variable or potential race conditions.